pkg/controller/batchrelease/batchrelease_plan_executor.go

```diff
@@ -242,21 +245,35 @@ func (r *Executor) progressBatches(workloadController workloads.WorkloadControll
 		verified, err := workloadController.CheckOneBatchReplicas()
 		switch {
 		case err != nil:
-			setCondition(r.releaseStatus, "VerifyBatchReadyError", err.Error(), v1.ConditionFalse)
+			setCondition(status, "Progressing", v1.ConditionFalse, "VerifyBatchFailed", err.Error())
 		case verified:
 			result = reconcile.Result{RequeueAfter: DefaultDuration}
-			r.releaseStatus.CanaryStatus.BatchReadyTime = metav1.Now()
-			r.releaseStatus.CanaryStatus.CurrentBatchState = v1alpha1.ReadyBatchState
+			status.CanaryStatus.BatchReadyTime = metav1.Now()
+			status.CanaryStatus.CurrentBatchState = v1alpha1.ReadyBatchState
 		default:
-			r.releaseStatus.CanaryStatus.CurrentBatchState = v1alpha1.InitializeBatchState
+			status.CanaryStatus.CurrentBatchState = v1alpha1.InitializeBatchState
 		}
 
 	case v1alpha1.ReadyBatchState:
 		klog.V(3).Infof("BatchRelease(%v) Batch State Machine into %s state", r.releaseKey, v1alpha1.ReadyBatchState)
-		// expected pods in the batch are upgraded and their state are ready
-		// wait to move to the next batch if there are any
-		progressDone = r.moveToNextBatch()
-		result = reconcile.Result{RequeueAfter: DefaultDuration}
+		if !r.isPartitioned() {
+			currentTimestamp := time.Now()
+			currentBatch := r.releasePlan.Batches[r.releaseStatus.CanaryStatus.CurrentBatch]
+			waitDuration := time.Duration(currentBatch.PauseSeconds) * time.Second
```

Comment:
is it possible that the user has not set PauseSeconds, it is currently defaulted to zero pause seconds, which is not proper since one may need manual confirmation

Reply:
BatchRelease use BatchPartition to support the feature of manual approve. For example, users set BatchPartition=2, BatchRelease will be paused when the batch 2 is ready.